Split release table by channel

steverydz opened the pull request:
## Done
Implemented mobile designs for the release UI

## QA
- Go to https://snapcraft-io-3990.demos.haus/core/releases
- Resize to mobile size
- Check that it matches [the designs](https://app.zeplin.io/project/61b3231e21194c4a48005226/screen/61ddabca6985e510c8505f0a)

## Issue
Fixes https://github.com/canonical-web-and-design/marketplace-tribe/issues/2481, https://github.com/canonical-web-and-design/marketplace-tribe/issues/2483

webteam-app commented:
Demo starting at https://snapcraft-io-3990.demos.haus

steverydz commented:
@anasereijo This is the final PR for the release UI mobile build so you can review the whole thing.

Lukewh commented on static/js/publisher/release/components/releasesTable/releaseCell.js:
```diff
@@ -92,6 +92,7 @@ const ReleasesTableReleaseCell = (props) => {
   };
 
   function handleHistoryIconClick(arch, risk, track, branchName) {
+    window.scrollTo(0, 0);
```
I always think when we do this we should store where the user came from (scroll) so when they close we can send them back - not for now, just to think about.

Lukewh reviewed (approved):
:+1: LGTM

anasereijo commented:
**- [ ] some design issues**
how it looks like:
![Screenshot 2022-05-16 at 16.47.26.png](https://images.zenhubusercontent.com/5a9531ab4b5806bc2bc88d30/561bb436-ff06-432a-a6ad-1790ffb7add9)

how it should look like:
![Screenshot 2022-05-16 at 16.47.10.png](https://images.zenhubusercontent.com/5a9531ab4b5806bc2bc88d30/9e2fb77d-3e21-4de5-a1a3-7107ccae38f8)

**- [ ] can we have the filter by track dropdown?**
how it looks like:
![Screenshot 2022-05-16 at 16.39.07.png](https://images.zenhubusercontent.com/5a9531ab4b5806bc2bc88d30/778ca200-b5d8-4486-be33-dfa48c5bf6a4)

how it should look like:
![Screenshot 2022-05-16 at 16.38.52.png](https://images.zenhubusercontent.com/5a9531ab4b5806bc2bc88d30/8e3d53c8-1af8-473a-ad0a-64b1141037de)

**- [ ] when clicking on a revision there are some design issues:** please check the [design](https://app.zeplin.io/project/61b3231e21194c4a48005226/screen/61e18c10f4e78e4fe0595ef9)
![Screenshot 2022-05-16 at 16.51.58.png](https://images.zenhubusercontent.com/5a9531ab4b5806bc2bc88d30/cce4a0c9-88c8-4d10-a159-1bbe712c4dc5)


**- [ ] is it possible to match the design for the 'show more' revisions?**
how it looks like:
![Screenshot 2022-05-16 at 16.59.28.png](https://images.zenhubusercontent.com/5a9531ab4b5806bc2bc88d30/225f9c20-09ea-4620-b673-b1fb2e4f7240)

how it should look like:
![Screenshot 2022-05-16 at 16.55.30.png](https://images.zenhubusercontent.com/5a9531ab4b5806bc2bc88d30/17808c0a-12c6-4904-ad3b-e9c0c05a117b)

steverydz commented:
@anasereijo I've addressed all of your points. I've made issues for the track filter and the pagination because neither of those are trivial changes.

https://github.com/canonical-web-and-design/snapcraft.io/issues/4000
https://github.com/canonical-web-and-design/snapcraft.io/issues/4001

anasereijo commented:
Thanks @steverydz
